### ISSUE TITLE
OOM in neofs-storage.service after load a big objects (4GB)

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#2003]
After install new environment and try to load a very big files, one of 4GB, and second of 4TB I saw a lot of OOM errors on logs on 2 nodes:
```
Nov 01 06:58:38 glagoli kernel: oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/system.slice/neofs-storage.service,task=neofs-node,pid=1177,uid=997
Nov 01 06:58:38 glagoli kernel: Out of memory: Killed process 1177 (neofs-node) total-vm:4255632kB, anon-rss:3392008kB, file-rss:0kB, shmem-rss:0kB, UID:997 pgtables:6828kB oom_score_adj:0
Nov 01 06:58:38 glagoli systemd[1]: neofs-storage.service: Main process exited, code=killed, status=9/KILL
Nov 01 06:58:38 glagoli vmagent[495]: 2022-11-01T06:58:38.228Z        warn        VictoriaMetrics/app/vmagent/remotewrite/client.go:277        couldn't send a block with size 1346 bytes to "1:secret-url": Post "http://localhost:8428/api/v1/write": dial tcp4: lookup localhost: i/o timeout; re-sending the block in 4.000 seconds
Nov 01 06:58:38 glagoli vmagent[495]: 2022-11-01T06:58:21.262Z        error        VictoriaMetrics/lib/promscrape/scrapework.go:355        error when scraping "http://localhost:8086/metrics" from job "node" with labels {instance="glagoli.neofs",job="node",type="neofs s3 gw"}: error when scraping "http://localhost:8086/metrics" with timeout 5s: timeout
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.237Z        INFO        new peer connected        {"addr": "172.26.163.47:40333", "peerCount": 2}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.163.47:40333", "peerCount": 3}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.163.47:40333", "peerCount": 4}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.162.119:40333", "peerCount": 5}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.160.213:40333", "peerCount": 6}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.161.216:40333", "peerCount": 7}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.161.216:39570", "peerCount": 8}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.162.119:40333", "peerCount": 9}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.239Z        INFO        new peer connected        {"addr": "172.26.160.213:40333", "peerCount": 10}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.242Z        WARN        peer disconnected        {"addr": "172.26.161.216:39570", "error": "identical node id", "peerCount": 9}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.245Z        WARN        peer disconnected        {"addr": "172.26.161.216:40333", "error": "unexpected empty payload: CMDVersion", "peerCount": 8}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.245Z        INFO        new peer connected        {"addr": "172.26.162.119:40333", "peerCount": 9}
Nov 01 06:58:38 glagoli neo-go[319]: 2022-11-01T06:58:38.245Z        INFO        new peer connected        {"addr": "172.26.163.47:40333", "peerCount": 10}
Nov 01 06:58:38 glagoli systemd[1]: neofs-storage.service: Failed with result 'signal'.
Nov 01 06:58:38 glagoli systemd[1]: neofs-storage.service: Consumed 11.265s CPU time.
```
![image](https://user-images.githubusercontent.com/110966367/199187374-20964bc3-9602-40ed-8c69-eb0d363a2e22.png)
![image](https://user-images.githubusercontent.com/110966367/199187388-bb13c24c-8b87-419b-b149-31a17319e332.png)

**Logs:**
[journal_az.tar.zip](https://github.com/nspcc-dev/neofs-node/files/9908287/journal_az.tar.zip)
[journal_buki.tar.zip](https://github.com/nspcc-dev/neofs-node/files/9908286/journal_buki.tar.zip)
[journal_vedi.tar.zip](https://github.com/nspcc-dev/neofs-node/files/9908288/journal_vedi.tar.zip)
[journal_glagoli.tar.zip](https://github.com/nspcc-dev/neofs-node/files/9908285/journal_glagoli.tar.zip)

**Versions:**
```
NeoFS Storage node
Version: v0.33.0-91-g98034005
GoVersion: go1.18.4
```

## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

**Comment by anikeev-yadro** [comment]
OOM reproduced on the autotest ```testsuites.network.test_node_management#test_add_nodes```
```
	Строка 112899: Nov 11 05:03:25 vedi kernel: prometheus-aler invoked oom-killer: gfp_mask=0x100cca(GFP_HIGHUSER_MOVABLE), order=0, oom_score_adj=0
	Строка 112905: Nov 11 05:03:25 vedi kernel:  oom_kill_process.cold+0xb/0x10
	Строка 112957: Nov 11 05:03:25 vedi kernel: [  pid  ]   uid  tgid total_vm      rss pgtables_bytes swapents oom_score_adj name
	Строка 113025: Nov 11 05:03:25 vedi kernel: oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/system.slice/neofs-storage.service,task=neofs-node,pid=16375,uid=998
	Строка 113026: Nov 11 05:03:25 vedi kernel: Out of memory: Killed process 16375 (neofs-node) total-vm:7301188kB, anon-rss:6530260kB, file-rss:0kB, shmem-rss:0kB, UID:998 pgtables:12952kB oom_score_adj:0
	Строка 113341: Nov 11 05:03:24 vedi systemd[1]: neofs-storage.service: A process of this unit has been killed by the OOM killer.
	Строка 113390: Nov 11 05:03:25 vedi systemd[1]: neofs-storage.service: Failed with result 'oom-kill'.
```
**Logs:** [41a5c8934e0353d4.zip](https://github.com/nspcc-dev/neofs-node/files/9989567/41a5c8934e0353d4.zip)
**Test log:** [consoleText.zip](https://github.com/nspcc-dev/neofs-node/files/9989605/consoleText.zip)

**Comment by fyrchik** [comment]
We have lots of similar messages in both logs, right before the OOM happens:
```
Nov 11 04:54:04 vedi neofs-node[16375]: 2022-11-11T04:54:02.887Z        debug        policer/check.go:219        shortage of object copies detected        {"component": "Object Policer", "object": "3uhTRAycFL5LozSVZ5RWzdacxfRc4cbCig4WFD98Ge44/57sDTnsRaoFeHweqSMQq5YAkrh8iMsbe929LufbRnQbU", "shortage": 1}
...
Nov 11 04:54:12 vedi neofs-node[16375]: 2022-11-11T04:54:11.248Z        error        replicator/process.go:62        could not replicate object        {"component": "Object Replicator", "node": "02614e526050e6aa619f29cd175d8cda48f0066186815eedf87a22d53927424a22", "object": "3uhTRAycFL5LozSVZ5RWzdacxfRc4cbCig4WFD98Ge44/57sDTnsRaoFeHweqSMQq5YAkrh8iMsbe929LufbRnQbU", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/node4.neofs/tcp/8080]: write object via client: rpc error: code = DeadlineExceeded desc = context deadline exceeded"}
```

1. We try to replicate each object.
2. Because the node is down we wait for the whole timeout.
3. During this time all objects persist in memory.
4. We quickly fill up replication pool with hanging goroutine each of which stores an object in memory.

**Comment by fyrchik** [comment]
Closed via #2049
